Hestia/app/src/main/java/hestia/UI/LoginActivity.java

```diff
@@ -122,4 +137,19 @@ private void editLoginAttempts(){
     private void showLoginToast(String info){
         Toast.makeText(getApplicationContext(),info,Toast.LENGTH_SHORT).show();
     }
+
+    public static String hashPassword(String password) {
+        MessageDigest digest = null;
+        String output = null;
+        try {
+            digest = MessageDigest.getInstance("SHA-512");
+            String input = password + SALT;
+            digest.update(input.getBytes("UTF-8"));
+            byte [] hash = digest.digest();
+            output = new String(hash, "UTF-8");
```

Comment:
Suggested reading: https://stackoverflow.com/questions/9212523/why-rehashing-will-make-it-more-secure